ui/dashboards/src/components/GridLayout/GridTitle.tsx

```diff
@@ -43,59 +43,63 @@ export function GridTitle(props: GridTitleProps) {
   const { openDeletePanelGroupDialog } = useDeletePanelGroupDialog();
   const { isEditMode } = useEditMode();
 
-  const text = (
-    <Typography variant="h2" sx={{ marginLeft: collapse !== undefined ? 1 : undefined }}>
-      {title}
-    </Typography>
-  );
+  const text = <Typography variant="h2">{title}</Typography>;
 
   return (
     <Box
+      onClick={collapse ? collapse.onToggleOpen : undefined}
       sx={{
         display: 'flex',
         justifyContent: 'start',
         alignItems: 'center',
         padding: (theme) => theme.spacing(1),
+        cursor: collapse ? 'pointer' : 'auto',
         backgroundColor: ({ palette }) =>
           palette.mode === 'dark' ? palette.background.paper : palette.background.default,
       }}
       data-testid="panel-group-header"
-      onClick={collapse !== undefined ? collapse.onToggleOpen : undefined }
-      style={{ cursor: 'pointer' }}
     >
       {collapse ? (
         <>
-          {collapse.isOpen ? <ExpandedIcon style={{ margin: '8px' }}/> : <CollapsedIcon style={{ margin: '8px' }}/>}
+          <IconButton sx={{ marginRight: 1 }} aria-label={`${collapse.isOpen ? 'collapse' : 'expand'} group ${title}`}>
+            {collapse.isOpen ? <ExpandedIcon /> : <CollapsedIcon />}
+          </IconButton>
           {text}
           {isEditMode && (
             <Stack direction="row" marginLeft="auto">
               <InfoTooltip description={TOOLTIP_TEXT.addPanelToGroup}>
-                <IconButton 
+                <IconButton
                   aria-label={ARIA_LABEL_TEXT.addPanelToGroup(title)}
                   onClick={(e) => {
-                    {if (collapse.isOpen) e.stopPropagation()}; // set as expanded if not already
+                    // Don't trigger expand/collapse
```

Comment:
I was proposing to expand the panel group (thus not stopping the event propagation) in case of this add panel action, because imho it feels weird when adding a new panel in a collapsed row and not seeing the result right away.

Actually thinking about it again it would be better to trigger this expand event only after saving the new panel instead.

Reply:
Yeah that sounds reasonable! Whatever we do, I figured this PR could be focused on only changing the row click behavior, and we can discuss any other change with designer + do in a separate PR

Reply:
Ok fair enough!